addons/account/models/account_move.py

```diff
@@ -2030,8 +2030,8 @@ def open_reconcile_view(self):
     def message_new(self, msg_dict, custom_values=None):
         # OVERRIDE
         # Add custom behavior when receiving a new invoice through the mail's gateway.
-        if custom_values.get('type', 'entry') not in ('out_invoice', 'in_invoice'):
-            return False
+        if not custom_values or custom_values.get('type', 'entry') not in ('out_invoice', 'in_invoice'):
```

Comment:
Little cosmetic remark but if you prefer your way, I'm ok:
`if not custom_values or custom_values.get('type', 'entry') not in ('out_invoice', 'in_invoice'):`
vs
`if (custom_values or {}).get('type', 'entry') not in ('out_invoice', 'in_invoice'):`

Also, you could simplify the next statement by:
`super().message_new(msg_dict, custom_values=custom_values)`